bool top_sensor
bool middle_sensor
bool bottom_sensor
int32 block_num
